src/modules/tags/TagDetail.js

```diff
@@ -16,6 +16,13 @@ const TagDetail = (props) => {
   const [useSpots] = useSpotsHook();
   const [useTags] = useTagsHook();
   const selectedTag = useSelector(state => state.project.selectedTag);
+  const [refresh, setRefresh] = useState(false);
+
+  useEffect(() => {
+    setRefresh(!refresh); // #TODO : Current hack to render two different FlatListComponents when selectedTag Changes.
```

Comment:
I'll look at this tomorrow to see if I can help you find a way that isn't a hack. Looks like @NathanNovak did a similar hack in CustomFeatureTypes.js and ActiveDatasetsList.js but that hack is a little cleaner. Check that out if we can find the correct way to fix this.

Reply:
I tried the hack used by @NathanNovak , it did not help. I guess the usecases are different in both the scenarios.